### PR TITLE
Test using mock

### LINKED ISSUE #106: Auth module is not implemented as interface-class
Now auth module only has Supabase based authentication and access control implementation. This logic is included as decorator methods directly in the module's `__init__.py`. Other modules follow the pattern of first defining an interface with methods signatures and common method implementations. Then these interfaces are implemented with separate technologies. 

Could change the Auth module also to have an InterfaceClass and Supabase as an implementation of this interface.

### PR DESCRIPTION
**Pull request by kavitharaju**
This PR Inlcudes
- Enhancements to some of the existing tests #79 

  - [x] Mock Supabase access to check token and role allowing dummy users to run tests
  - [x] Fixture to clean up postgres test DB after each test
  - [x] Dataupload tests
    - [x]  Run the 3 dataupload API tests on chroma and postgres alike
    - [ ]  Run text data upload with different fileprocessors, Langchain and vanilla (test written. But vanilla option erroring out)
    - [ ]  Run dataupload with different embeddings(Mock the ones that need paid services)
  - [x] Get labels API
    - [x] Test get labels works with changing data and different databases
  - [x]  Run tests automatically on PRs and pushes on github actions
  :warning: More tasks break down added in [issue](https://github.com/BibleNLP/assistant.bible/issues/79)
- Refactor of Auth module codebase to fix #106 
   - Needed to split up the decorator functions to be able to mock them. Along with that did the change to interface-class pattern keeping the existing logics intact. (tested only manually)

### CODE REVIEW

File: app/core/llm_framework/openai_langchain.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ class LangchainOpenAI(LLMFrameworkInterface):
     chain = None
     vectordb = None
     def __init__(self, #pylint: disable=super-init-not-called
-                key:str=os.getenv("OPENAI_API_KEY"),
```
Comment (ryderwishart) [inline app/core/llm_framework/openai_langchain.py:28]: I wonder if it might make sense to just have a `env.test` file that we load?

**Review by ryderwishart — APPROVED** [review]
Awesome work @kavitharaju. Looks great!

File: app/core/llm_framework/openai_langchain.py (hunk 1 above)
Comment (ryderwishart) [reply app/core/llm_framework/openai_langchain.py:28]: This is the only one I would think about, but it's not a big blocker if we don't want to do it now. Maybe just add a FIXME or something if not.

Comment (kavitharaju) [reply app/core/llm_framework/openai_langchain.py:28]: This is something I struggled with a lot...
What I actually tried to do was to mock the constructors of OpenAI and Supabase classes where they connect to the API service. This way, even if we have an API key saved in our env, we wont be charged accidentaly while running tests. But  for some reason couldn't get it to work. (I am using mock for the first time. )

I want to come back to it if I can... Will add a fixme.

File: app/routers.py [hunk 2]
```diff
@@ -477,12 +480,12 @@ async def signup(
...
     """Signs up a new user"""
     try:
-        data = supa.auth.sign_up({
+        data = auth_service.conn.auth.sign_up({
             "email": email, 
             "password": password,
             })
-    except gotrue.errors.AuthApiError as e:
-        raise PermissionException("Sign up error") from e
+    except gotrue.errors.AuthApiError as exe:
```
[the remark this answers is not in the document]
Comment (kavitharaju) [reply app/routers.py:487]: Yes, that is what I meant. Pylint suggest we use minimun 3 letter variable names.

**Pull request merged**